SDIAlliance/green-data-center-map
# [Feature] Data Center Map

## Task

[Notion Task](https://www.notion.so/concealed/Green-Data-Center-Map-6b1a1505f6c94b9ea69c137e462b77ed)

## Implementation

- I used the `web` folder of the [electricitymap repo](https://github.com/SDIAlliance/electricitymap-contrib) to implement the required features.
- Right now, I'm allowing the user to add the same Data Center Facility to the comparison panel, as the API only returns 1 entry.
- I'm requesting a different data center example on refresh. If you refresh the page, you will see that the data center changes location.
- I tried to reuse the already implemented logic as much as possible and make it work for our use case.

## Test
Go to Map
 - As soon as you see the map, find the orange circle that is present. It can be in any position of the map, including the ocean.
  - If you hover the orange circle, you will see the data center details on the tooltip.
  - When you click on it, the left panel will open and the data center will be added to the data center comparison.
  - Click on the data center one more time in order to add it again. 
  - Now the "compare" button is enabled.
  - Press on the compare button to expand the panel and compare the data centers.
  - A table will display with the data centers names and info.
  - If you click "go back", you will see the selected data centers list again. 
  - Remove data centers until you only have one and the "compare" button will be disabled again.
  - If you click on the other panel tab you will see the default electricity map panel. You can switch between the two anytime or close the panel by clicking on the currently open toggle.
